DependencyInjection/CmfResourceExtension.php

```diff
@@ -48,6 +51,37 @@ public function load(array $configs, ContainerBuilder $container)
         $loader->load('resource.xml');
 
         $this->loadRepositories($container, $config['repositories']);
+        $this->loadDescriptionEnhancers($container, $config['description']['enhancers']);
+    }
+
+    /**
+     * Add a repository DI definition factory.
+     *
+     * @param string $name
+     * @param RepositoryFactoryInterface $factory
+     */
+    public function addRepositoryFactory($name, RepositoryFactoryInterface $factory)
+    {
+        $this->repositoryFactories[$name] = $factory;
+    }
+
+    /**
+     * Add a description enhancer DI definition factory.
+     *
+     * @param string $name
+     * @param EnhancerFactoryInterface
+     */
+    public function addDescriptionEnhancerFactory($name, EnhancerFactoryInterface $enhancer)
+    {
+        $this->descriptionEnhancerFactories[$name] = $enhancer;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getNamespace()
+    {
+        return 'http://cmf.symfony.com/schema/dic/'.$this->getAlias();
```

Comment:
while this isn't officially part of the PR, using `getAlias()` is useless: We'll never change the alias and if we do, we have to do search & replace anyway because it's used in many static cases.